runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java

```diff
@@ -202,10 +203,21 @@ else if (config.lvl() && message.toLowerCase().equals(CMB_COMMAND_STRING))
 		}
 		else if (config.price() && message.toLowerCase().startsWith(PRICE_COMMAND_STRING + " "))
 		{
-			String search = message.substring(PRICE_COMMAND_STRING.length() + 1);
-
-			log.debug("Running price lookup for {}", search);
-			executor.submit(() -> itemPriceLookup(setMessage.getMessageNode(), search));
+			Matcher priceMatcher = PRICE_PATERN.matcher(message.toLowerCase());
```

Comment:
We already know the string begins with `!price` here so you can just match `search` against `(?:(\\d*) )?(.+)`